python/tests/mlir/bool_var_scope.py

```diff
@@ -0,0 +1,35 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+# RUN: PYTHONPATH=../../ pytest -rP  %s | FileCheck %s
+
+import cudaq
+
+import pytest
+
+def test_var_scope():
+
+    @cudaq.kernel
+    def kernel():
+        qubits = cudaq.qvector(2)
+        var_int = 42
+        var_bool = True
+
+    print(kernel)
+
+
+# CHECK-LABEL:   func.func @__nvqpp__mlirgen__kernel() attributes {"cudaq-entrypoint"} {
+# CHECK:           %[[VAL_0:.*]] = arith.constant true
+# CHECK:           %[[VAL_1:.*]] = arith.constant 42 : i64
+# CHECK:           %[[VAL_2:.*]] = quake.alloca !quake.veq<2>
+# CHECK:           %[[VAL_3:.*]] = cc.alloca i64
+# CHECK:           cc.store %[[VAL_1]], %[[VAL_3]] : !cc.ptr<i64>
+# CHECK:           %[[VAL_4:.*]] = cc.alloca i1
```

Comment:
All the variables should be inserted at the top of the entry block, so this alloca, store, alloca pattern would ought to look like alloca, alloca, store, ...

Reply:
This seems like a larger issue, now captured in https://github.com/NVIDIA/cuda-quantum/issues/1963